### DISCUSSION TITLE
Custom routes not working

### DISCUSSION THREAD

**Discussion by arthurvmdantas** [pocketbase/pocketbase#5027] [Q&A] [answered]
Reading the docs, adding a custom route seems to be a simple task but I'm unable to call my custom route. It's defined inside the file `pb_hooks/main.pb.js` with the content

```

routerAdd("GET", "/custom", (c) => {
  return c.json(200, { "message": "Custom"});
});
```

I'm trying to call it using a REST client with the `http://127.0.0.1:8090/api/custom` endpoint, but every response is an HTTP 404.

To make a test, I added a custom middleware in the same file and it worked as expected. 

```
routerUse((next) => {
  return (c) => {
    console.log("inside middleware");
    return next(c);
  };
});
```

PocketBase is even restarting when I modify the file. Hooks are also working.

Maybe this is related to permissions or I'm calling the wrong endpoint? I'm running PocketBase (0.22.12) on Mac OS.

**Comment by ganigeorgiev** [comment, marked as answer]
We don't prepend `/api/`. The resulting route path is exactly the same as in the route definition, or in your case: `GET http://127.0.0.1:8090/custom`

**Comment by arthurvmdantas** [comment]
Thanks! Maybe this information should be more explicit in the documentation, perhaps with an example of the endpoint created (`http://127.0.0.1:8090/hello`).

Instead, currently, right after the route example code (hello route), there is an info box showing a route with "/api/...". I should've paid more attention, but it got stuck in my head (English is not my first language). It's a thought.

Thanks again.
